examples/apps/src/ble_bas_peripheral.rs

```diff
@@ -106,20 +107,22 @@ async fn gatt_events_task<P: PacketPool>(server: &Server<'_>, conn: &GattConnect
     let reason = loop {
         match conn.next().await {
             GattConnectionEvent::Disconnected { reason } => break reason,
-            GattConnectionEvent::Gatt { event: Err(e) } => warn!("[gatt] error processing event: {:?}", e),
-            GattConnectionEvent::Gatt { event: Ok(event) } => {
-                match &event {
-                    GattEvent::Read(event) => {
-                        if event.handle() == level.handle {
+            GattConnectionEvent::Gatt { event } => {
+                match event.payload().incoming() {
```

Comment:
This is a little bit footgun of an API in my mind. For a a GattConnection, the payload will always be incoming, so it seems wrong to allow choosing to do anything else than treat it as a AttClient::Request.

Reply:
I'm not sure I understand as to where the problem is?

For a `GattConnection`, the payload will always be incoming sure, and there is anyway no "outgoing" method on `GattData` if that's the issue.

If the issue is that the user can match not just on `AttClient::Request` but also on `AttClient::Command` or `AttClient::Confirmation`, this is IMO exactly as it should be, because incoming (client -> server) ATT payloads are _not_ just "requests". You might receive commands too, as well as confirmations. In fact, `rs-matter`'s custom logic matches on `AttClient::Confirmation`.

Reply:
Topic no longer applicable after the switch to the `GattEvent::Other` approach.